templates/demo/web3js-react-dapp-demo/src/App.tsx

```diff
@@ -0,0 +1,136 @@
+      setChainId(BigInt(newId));
+    }
+
+    web3Context.web3.eth.getChainId().then(updateChainId);
+    provider.on("chainChanged", updateProviderChainId);
+    return () => provider.removeListener("chainChanged", updateProviderChainId);
+  }, [web3Context.currentProvider, web3Context.web3.eth]);
+
+  return (
+    <main>
+      <h1>Web3.js + React Minimal dApp Template</h1>
+      <div>
+        This is a sample project that demonstrates using{" "}
+        <a href="https://web3js.org/">Web3.js</a> with the{" "}
+        <a href="https://react.dev/">React</a> front-end framework.
+        <ul>
+          <li>
+            <a href="https://docs.web3js.org/">Web3.js Docs</a>
+          </li>
+          <li>
+            <a href="https://react.dev/learn">React Docs</a>
+          </li>
+        </ul>
+        Learn more about this project and its design by referring to the{" "}
+        <a href="https://github.com/web3/create-web3js-dapp/blob/main/templates/demo/web3js-react-dapp-demo/README.md">
+          GitHub README
+        </a>
+        .
+      </div>
+      {!hasProviders ? (
+        <>
+          <h2>Install a Wallet</h2>
+          <div>Install a wallet browser extension:</div>
+          <ul>
+            <li>
+              <a href="https://www.enkrypt.com/download.html">Enkrypt</a>
+            </li>
+            <li>
+              <a href="https://www.exodus.com/download/">Exodus</a>
+            </li>
+            <li>
+              <a href="https://metamask.io/download/">MetaMask</a>
+            </li>
+            <li>
+              <a href="https://trustwallet.com/download">Trust Wallet</a>
+            </li>
+          </ul>
+        </>
+      ) : web3Context.currentProvider === undefined ? (
+        <>
+          <h2>Select a Provider</h2>
+          {web3Context.providers.map(
+            (provider: providers.EIP6963ProviderDetail) => {
+              return (
+                <div key={provider.info.uuid}>
+                  <ProviderButton provider={provider}></ProviderButton>
+                </div>
+              );
+            },
+          )}
+        </>
+      ) : (
+        <>
+          {web3Context.providers.length > 1 ? (
+            <>
+              <h2>Switch Provider</h2>
+              {web3Context.providers.map(
+                (provider: providers.EIP6963ProviderDetail) => {
+                  if (
+                    provider.info.uuid ===
+                    web3Context.currentProvider?.info.uuid
+                  ) {
+                    return null;
+                  }
+                  return (
+                    <div key={provider.info.uuid}>
+                      <ProviderButton provider={provider}></ProviderButton>
+                    </div>
+                  );
+                },
+              )}
+            </>
+          ) : null}
+          <h2>Network Details</h2>
+          <div>Chain ID: {`${chainId}`}</div>
```

Comment:
Showing networkid would be cool too :)

Reply:
How does one determine that? All I can find is `web3.eth.defaultNetworkId`. Is that it?

Reply:
await web3.eth.net.getId();